Source/SIMPLib/Geometry/ImageGeom.cpp

```diff
@@ -675,6 +675,46 @@ void ImageGeom::getCoords(size_t idx, double coords[3]) const
   coords[2] = static_cast<double>(plane) * m_Spacing[2] + m_Origin[2] + (0.5 * m_Spacing[2]);
 }
 
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+std::optional<size_t> ImageGeom::getIndex(float coords[3]) const
+{
+  return getIndex(coords[0], coords[1], coords[2]);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+std::optional<size_t> ImageGeom::getIndex(float xCoord, float yCoord, float zCoord) const
+{
+  size_t x = (xCoord - (0.5 * m_Spacing[0]) - m_Origin[0]) / m_Spacing[0];
+  size_t y = (yCoord - (0.5 * m_Spacing[1]) - m_Origin[1]) / m_Spacing[1];
+  size_t z = (zCoord - (0.5 * m_Spacing[2]) - m_Origin[2]) / m_Spacing[2];
```

Comment:
Despite passing in floats, this method is performing double-precision when calculating x, y, and z. I normally would not say anything about such matters, but this method is a completely identical to `getIndex(double, double, double)`. Either stay consistent to the parameter precision or consolidate calculations into a single method. Copy-pasting should be avoided.

```suggestion
  size_t x = (xCoord - (0.5f * m_Spacing[0]) - m_Origin[0]) / m_Spacing[0];
  size_t y = (yCoord - (0.5f * m_Spacing[1]) - m_Origin[1]) / m_Spacing[1];
  size_t z = (zCoord - (0.5f * m_Spacing[2]) - m_Origin[2]) / m_Spacing[2];
```